develop/tutorials/articles/340-data-upgrades-and-verifiers/creating-an-upgrade-process-for-your-application.markdown

```diff
@@ -202,16 +203,16 @@ Here are some organizational tips:
 -   Group together similar database updates (ones that operate on a data element
     or related data elements) in the same upgrade step class.
 
--   Create upgrade steps in packages named after each data schema version. 
+-   Create upgrade steps in sub-packages named after each data schema version. 
 
 After creating upgrade steps, you can create a registrator that associates them
 with their target schema versions. 
 
-### Writing the Upgrade Step Registrator [](id=writing-the-upgrade-step-registrator)
+## Writing the Upgrade Step Registrator [](id=writing-the-upgrade-step-registrator)
 
-The upgrade step registrator (registrator) is responsible for registering
-upgrade steps with each of the module's schema versions. It orchestrates the
-module's entire upgrade process. 
+The upgrade step registrator (registrator) registers upgrade steps with each
+of the module's schema versions. It orchestrates the module's entire upgrade
+process.
```

Comment:
What does it mean to register upgrade steps? Does this involve notifying some framework? With what (Upgrade Framework) are the steps registered? For example, does the Upgrade Framework have a registry for upgrade steps to execute? Who controls executing the registered upgrade steps and how is the order of step execution determined? When are the steps executed?